packages/components/Button/src/Button.types.ts

```diff
@@ -110,7 +111,7 @@ export interface IButtonProps extends Omit<IPressableProps, 'onPress'> {
 export interface IButtonSlotProps {
   root: React.PropsWithRef<IViewWin32Props>;
   stack: ViewProps;
-  icon: ImageProps;
+  icon: string | number | IconProps;
```

Comment:
This should be the prop type of the component hosted in this slot. So if this is an Icon component it should be IconProps. It isn't related to the icon prop on the public interface.

Reply:
This is the only comment that I would consider blocking, as it could lead to unexpected bugs like a number getting passed as props into the icon component, which would...well I have no idea, but it likely wouldn't work as expected.

Reply:
The issue with that is when you do something like require('.test.png'), what is returned is actually a number that is an index into the AssetRegistry.

Reply:
But this is the prop type of the internal component. This is the equivalent of passing the a number instead of a props interface to the Icon component. It is not the type for the icon prop on the button props interface.

Reply:
Oh I see this is about the slot props. Ok its removed now.